src/server/game/Entities/Player/Player.cpp

```diff
@@ -22069,17 +22069,17 @@ bool Player::ActivateTaxiPathTo(std::vector<uint32> const& nodes, Creature* npc
     // taximaster case
     if (npc)
     {
+
         // not let cheating with start flight mounted
         if (IsMounted())
         {
-            GetSession()->SendActivateTaxiReply(ERR_TAXIPLAYERALREADYMOUNTED);
-            return false;
+            Dismount();
+            RemoveAurasByType(SPELL_AURA_MOUNTED);
         }
 
-        if (IsInDisallowedMountForm())
+        if (GetDisplayId() != GetNativeDisplayId())
```

Comment:
This isn't correct, transforms that allow mounting will not be removed
What about transform auras expiring mid-flight when there is more than one transform aura remaining?

Reply:
Didn't think about that, what will be the correct form then?

Reply:
I don't really know how to do that because I can't just remove the aura since it needs to be reapplied when the path finishes.